src/rgw/rgw_s3select.cc

```diff
@@ -460,6 +463,58 @@ int RGWSelectObj_ObjStore_S3::run_s3select_on_parquet(const char* query)
   return status;
 }
 
+int RGWSelectObj_ObjStore_S3::run_s3select_on_json(const char* query, const char* input, size_t input_length)
+{
+    int status = 0;
+  
+    uint32_t length_before_processing, length_post_processing;
+    const char* s3select_resource_id = "resourcse-id";
+    const char* s3select_processTime_error = "s3select-ProcessingTime-Error";
+    
+    m_aws_response_handler.init_response();
+    if (input == nullptr) {
+      input = "";
+    }
+    std::string temp_result;
+    m_aws_response_handler.init_success_response();
+    length_before_processing = temp_result.size();
+    //query is correct(syntax), processing is starting.
+    status = m_s3_json_object.run_s3select_on_stream(temp_result, input, input_length, s->obj_size);
+    m_aws_response_handler.get_sql_result().append(temp_result);
```

Comment:
ok.
`temp_result` implies it's a temporary resolution.
upon this issue (override the SQL result) is resolved, this code should be re-factored.